File: WorldFeed/Server/src/WorldFeed.Common/DTO/CreateMessageDTO.cs
namespace WorldFeed.Common.DTO
{
    using Newtonsoft.Json;

    using WorldFeed.Common.JsonConverters;
    using WorldFeed.Common.Public.Models.Interfaces.DTO;
    using WorldFeed.Common.Public.Models.Interfaces.DTO.Events;

    public class CreateMessageDTO : ICreateMessageDTO
    {
        [JsonProperty("event")]
        [JsonConverter(typeof(JsonPropertyConverterRepository))]
        public IMessageEventDTO MessageEvent { get; set; }
    }
}

File: Sharebook/Client/src/app/core/Core/DTO/CreateMessageDTO.ts
import {IMessageEventDTO} from "../../Public/Models/Interfaces/DTO/Events/IMessageEventDTO";
import {ICreateMessageDTO} from "../../Public/Models/Interfaces/DTO/ICreateMessageDTO";

export class CreateMessageDTO implements ICreateMessageDTO {
  // [JsonProperty("event")]
  // [JsonConverter(typeof(JsonPropertyConverterRepository))]
  public messageEvent: IMessageEventDTO;
}
